web/packages/teleterm/src/services/fileStorage/fileStorage.ts

```diff
@@ -91,19 +110,38 @@ export function createFileStorage(opts: {
     get,
     replace,
     getFilePath,
+    getFileLoadingError,
   };
 }
 
-function loadState(filePath: string) {
+async function loadState(
+  filePath: string
+): Promise<{ state: any; error: Error | undefined }> {
   try {
-    if (!existsSync(filePath)) {
-      writeFileSync(filePath, '{}');
-    }
-
-    return JSON.parse(readFileSync(filePath, { encoding: 'utf-8' }));
+    const file = await readOrCreateFile(filePath);
+    return {
+      state: JSON.parse(file),
```

Comment:
Also, it's a shame that Node.js doesn't have a dedicated function for parsing JSON files because the errors `JSON.parse` returns are rather unhelpful. It mentions the position in the string rather than the file which would be much more helpful.

Reply:
True :(